### DISCUSSION TITLE
"entity not found" 404 when trying to add a name to a face

### DISCUSSION THREAD

**Discussion by hkelly93** [photoprism/photoprism#2201] [Q&A]
Hello, I'm running photoprism in docker, and it's amazing so far...except I can only label one person. When I try to label another, I get an "entity not found" toast and the following in the console

<img width="429" alt="image" src="https://user-images.githubusercontent.com/11215222/160495967-74c26b10-81a3-4969-99a1-8858e5ff9cfd.png">

<img width="360" alt="image" src="https://user-images.githubusercontent.com/11215222/160495991-7d626b89-2b6f-4774-b3f0-31270f84b1fe.png">

I'm not sure what else to try. I've logged in/out, restarted the container, and tried it on a different device.

Any idea what could cause this or how I could look into it more? Like I said, the first person I labeled worked and found all the pictures correctly.

Thanks!

**Comment by graciousgrey** [comment]
Could you turn on debug logs (https://docs.photoprism.app/user-guide/settings/advanced/#debug-logs) and see what logs and errors you see inside PhotoPrism?

**Reply by xpil** [reply]
I had the same issue. Enabling the debug flag fixed it. And disabling the flag did not bring the issue back.
This is a heisenbug.

**Reply by hkelly93** [reply]
Yes, sorry for the delayed response!

This is what the logs show me

2022-03-30 17:46:43.513386+00:00time="2022-03-30T13:46:43-04:00" level=info msg="http: starting web server at 0.0.0.0:2342"
2022-03-30 17:46:56.468229+00:00time="2022-03-30T13:46:56-04:00" level=debug msg="http: GET / (200) [14.527949ms]"
2022-03-30 17:46:56.709815+00:00time="2022-03-30T13:46:56-04:00" level=debug msg="photos: found 7 results for public:true dist:20 year:0 month:0 quality:3 count:60 order:newest merged:true [7.76637ms]"
2022-03-30 17:46:56.727576+00:00time="2022-03-30T13:46:56-04:00" level=debug msg="http: GET /api/v1/photos?count=60&offset=0&merged=true&country=&camera=0&lens=0&label=&year=0&month=0&color=&order=newest&q=&public=true&quality=3 (200) [37.512877ms]"
2022-03-30 17:47:13.168424+00:00time="2022-03-30T13:47:13-04:00" level=debug msg="http: GET /api/v1/subjects?count=60&offset=0&q=&hidden=&order=relevance&files=1&type=person (200) [1.949822ms]"
2022-03-30 17:47:14.391799+00:00time="2022-03-30T13:47:14-04:00" level=debug msg="http: GET /api/v1/faces?count=999&offset=0&q=&hidden=&order=samples&markers=true&unknown=true (200) [893.096µs]"
2022-03-30 17:47:18.401195+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="api: abort /api/v1/markers/:marker_uid with code 404 (Entity not found)"
2022-03-30 17:47:18.401222+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="marker: record not found (find)"
2022-03-30 17:47:18.401227+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="http: PUT /api/v1/markers/mr9feq93td6mx0eg (404) [1.674467ms]"
2022-03-30 17:47:18.469731+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="api: abort /api/v1/markers/:marker_uid with code 404 (Entity not found)"
2022-03-30 17:47:18.469768+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="marker: record not found (find)"
2022-03-30 17:47:18.469775+00:00time="2022-03-30T13:47:18-04:00" level=debug msg="http: PUT /api/v1/markers/mr9feq93td6mx0eg (404) [1.525732ms]"
2022-03-30 17:47:20.274487+00:00time="2022-03-30T13:47:20-04:00" level=debug msg="http: GET /manifest.json?1646257436 (200) [2.850693ms]"

**Reply by lastzero** [reply]
Perhaps you have changed the face/person assignment of a file that has been deleted or archived, or is missing from the file system?

Commit https://github.com/photoprism/photoprism/commit/4afd3f2ca95b7500c2bfe7eaf19b457e62ddeffb adds more detailed error logs that also include the specific UID of the marker/file in question.

With this, you could check the database to see what's wrong with the file.... I would need more details on how to reproduce the problem so we can implement a solution if needed.

**Reply by aldobranti** [reply]
is it possible, that turning off settings/people and turning it back on again has confused the system ?

**Reply by phillip-simons** [reply]
> Perhaps you have changed the face/person assignment of a file that has been deleted or archived, or is missing from the file system?
> 
> Commit [4afd3f2](https://github.com/photoprism/photoprism/commit/4afd3f2ca95b7500c2bfe7eaf19b457e62ddeffb) adds more detailed error logs that also include the specific UID of the marker/file in question.
> 
> With this, you could check the database to see what's wrong with the file.... I would need more details on how to reproduce the problem so we can implement a solution if needed.

I have the same problem after this exact turn of events. There are a bunch of photos categorized under the person, but my guess is since the original photo it identified is gone there is some funkyness when trying to add the person

**Reply by lastzero** [reply]
Does it help to re-index? If it is because the file system has changed, for example an image has disappeared, then this may fix the index and the error.

**Comment by aldobranti** [comment]
If I can get the order right here, I was getting 'entity not found' from attempting to name a face in the 'new' tab
I then clicked on the image in new and selected (pen icon) to edit and followed to 'people' tab
I named the individual. Face recognition then took over across the images individual appeared in  and then the main sidebar/people tab knew about her

here are the logs

`time="2022-06-14T16:12:03Z" level=debug msg="http: GET /api/v1/photos?count=60&offset=0&merged=true&country=&camera=0&lens=0&label=&year=0&month=0&color=&order=newest&q=face:APMVRLPBNA7UFMTJ3YU3UDX2X37CT6VL&public=true&quality=0 (200) [39.083427ms]"
time="2022-06-14T16:12:10Z" level=debug msg="http: GET /api/v1/t/09800b9e67e5a890d429d2adcb8691decf5ab127/359a67f3/fit_1280 (200) [16.788711ms]"
time="2022-06-14T16:12:10Z" level=debug msg="http: GET /api/v1/t/a8d62eea782e61cde088514faecb9ddccba37e1d/359a67f3/fit_1280 (200) [29.938407ms]"
time="2022-06-14T16:12:10Z" level=debug msg="http: GET /api/v1/t/f4b25112fe137474bb62a991e38afd824ae062ee/359a67f3/fit_1280 (200) [41.020581ms]"
time="2022-06-14T16:12:12Z" level=debug msg="http: GET /api/v1/svg/photo (200) [74.74µs]"
time="2022-06-14T16:12:12Z" level=debug msg="http: GET /api/v1/photos/prc1elh1b6j8kpxp (200) [18.245188ms]"
time="2022-06-14T16:12:15Z" level=debug msg="crop: 09800b9e67e5a890d429d2adcb8691decf5ab127_2048x2048_fit.jpg is too small, upscaling 204px to 320px"
time="2022-06-14T16:12:15Z" level=debug msg="saved 09800b9e67e5a890d429d2adcb8691decf5ab127_320x320_crop_1a4148064085.jpg"
time="2022-06-14T16:12:15Z" level=debug msg="http: GET /api/v1/t/09800b9e67e5a890d429d2adcb8691decf5ab127-1a4148064085/359a67f3/tile_320 (200) [233.638405ms]"
time="2022-06-14T16:12:27Z" level=info msg="subject: added person 'Paula Sturges'"
time="2022-06-14T16:12:27Z" level=debug msg="covers: updated 14 subjects [6.810817ms]"
time="2022-06-14T16:12:27Z" level=debug msg="counts: updated 28 subjects [21.241473ms]"
time="2022-06-14T16:12:27Z" level=debug msg="photo: uid prc1elh1b6j8kpxp has new title 'Paula Sturges / 2017' [1.757768ms]"
time="2022-06-14T16:12:27Z" level=debug msg="photos: found 1 result for uid:prc1elh1b6j8kpxp merged:true [2.175391ms]"
time="2022-06-14T16:12:27Z" level=info msg="changes successfully saved"
time="2022-06-14T16:12:27Z" level=debug msg="http: PUT /api/v1/markers/mrc1elhrux9n24bq (200) [136.300379ms]"
time="2022-06-14T16:12:52Z" level=debug msg="http: GET /api/v1/subjects?count=60&offset=0&q=&hidden=&order=relevance&files=1&type=person (200) [4.666166ms]"
time="2022-06-14T16:13:00Z" level=debug msg="photos: found 10 results for public:true dist:20 year:0 month:0 subject:\"paula-sturges\" count:60 order:newest merged:true [50.93255ms]"
time="2022-06-14T16:13:00Z" level=debug msg="http: GET /api/v1/photos?count=60&offset=0&merged=true&country=&camera=0&lens=0&label=&year=0&month=0&color=&order=newest&q=person:paula-sturges&public=true&quality=0 (200) [92.157091ms]"
time="2022-06-14T16:16:30Z" level=debug msg="http: GET /api/v1/subjects?count=60&offset=0&q=&hidden=&order=relevance&files=1&type=person (200) [7.041072ms]"
time="2022-06-14T16:18:05Z" level=debug msg="http: GET /api/v1/config (200) [113.878481ms]"`

**Reply by aldobranti** [reply]
sorry about the formatting
